[core]/esx_multicharacter/server/main.lua

```diff
@@ -1,13 +1,18 @@
 if not ESX then
 	error('\n^1Unable to start Multicharacter - you must be using ESX Legacy^0')
 elseif ESX.GetConfig().Multichar then
+	local DatabaseConnected = false
```

Comment:
This full logic should not be used here because if there is no database connection then es_extended cannot retrieve the data from the database so it is useless to look elsewhere to see what is happening with the database. This should be tested in es_extended, but even oxmysql handles the sql connection so even there you don't need to just make sure that if there is no database connection then the code doesn't run.

Reply:
that isnt what this code is for :)

This is for selecting all tables that use owner and/or identifier which is required for multicharacter :)

Reply:
its an automated version of what the original from esx_kashacters (what esx_multicharacter is a fork of). in the original you had to put all of them into the config manually.

Reply:
Hmm interesting but why do we have to search at this level to use the database connection string? If you have the connection you can query the database to see if there is a table or not with a plain inform_ default sql table.

Reply:
Because normal sql queries with oxmysql act within the database, we need the data from outside of the database, to get table names and there columns.

Reply:
> Because normal sql queries with oxmysql act within the database, we need the data from outside of the database, to get table names and there columns.

It's query get current database but okey  SELECT DATABASE(); No problem. Thanks for describing why need this.